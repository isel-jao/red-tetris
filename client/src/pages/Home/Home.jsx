import React, { useState } from 'react';
import Button from '../../components/Button/Button';

/**
 * Home page component for the Red Tetris application
 * Allows users to create or join games
 */
const Home = () => {
  const [playerName, setPlayerName] = useState('');
  const [gameId, setGameId] = useState('');
  
  const handleCreateGame = () => {
    // Logic to create a new game
    console.log('Creating game with player:', playerName);
  };
  
  const handleJoinGame = () => {
    // Logic to join existing game
    console.log('Joining game:', gameId, 'with player:', playerName);
  };
  
  return (
    <div className="flex flex-col items-center justify-center min-h-screen bg-gray-100 p-4">
      <h1 className="text-4xl font-bold mb-8 text-center text-red-600">RED TETRIS</h1>
      
      <div className="bg-white rounded-lg shadow-lg p-6 w-full max-w-md">
        <div className="mb-4">
          <label htmlFor="playerName" className="block text-sm font-medium text-gray-700 mb-1">
            Your Name
          </label>
          <input
            type="text"
            id="playerName"
            value={playerName}
            onChange={(e) => setPlayerName(e.target.value)}
            className="w-full p-2 border border-gray-300 rounded focus:outline-none focus:ring-2 focus:ring-blue-500"
            placeholder="Enter your name"
            data-testid="player-name-input"
          />
        </div>
        
        <div className="flex space-x-2 mb-6">
          <Button 
            onClick={handleCreateGame}
            disabled={!playerName}
            data-testid="create-game-button"
            className="flex-1"
          >
            Create Game
          </Button>
        </div>
        
        <div className="relative">
          <div className="absolute inset-0 flex items-center">
            <div className="w-full border-t border-gray-300"></div>
          </div>
          <div className="relative flex justify-center text-sm">
            <span className="px-2 bg-white text-gray-500">or join existing game</span>
          </div>
        </div>
        
        <div className="mt-6 mb-4">
          <label htmlFor="gameId" className="block text-sm font-medium text-gray-700 mb-1">
            Game ID
          </label>
          <input
            type="text"
            id="gameId"
            value={gameId}
            onChange={(e) => setGameId(e.target.value)}
            className="w-full p-2 border border-gray-300 rounded focus:outline-none focus:ring-2 focus:ring-blue-500"
            placeholder="Enter game ID"
            data-testid="game-id-input"
          />
        </div>
        
        <Button 
          variant="secondary" 
          onClick={handleJoinGame}
          disabled={!playerName || !gameId}
          className="w-full"
          data-testid="join-game-button"
        >
          Join Game
        </Button>
      </div>
    </div>
  );
};

export default Home;
