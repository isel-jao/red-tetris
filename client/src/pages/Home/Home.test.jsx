import React from 'react';
import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import Home from './Home';

// Mock the Button component to simplify testing
vi.mock('../../components/Button/Button', () => ({
  default: ({ children, onClick, disabled, className, 'data-testid': testId }) => (
    <button 
      onClick={onClick} 
      disabled={disabled} 
      className={className} 
      data-testid={testId}
    >
      {children}
    </button>
  ),
}));

describe('Home Page', () => {
  // Test using react-test-renderer for snapshot testing
  it('renders correctly', () => {
    const tree = TestRenderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Test using React Testing Library for interaction
  it('updates player name when input changes', () => {
    render(<Home />);
    const input = screen.getByTestId('player-name-input');
    
    fireEvent.change(input, { target: { value: 'Player1' } });
    
    expect(input.value).toBe('Player1');
  });

  it('enables create game button when player name is entered', () => {
    render(<Home />);
    const button = screen.getByTestId('create-game-button');
    const input = screen.getByTestId('player-name-input');
    
    // Initially button should be disabled
    expect(button).toBeDisabled();
    
    // Enter player name
    fireEvent.change(input, { target: { value: 'Player1' } });
    
    // Button should now be enabled
    expect(button).not.toBeDisabled();
  });

  it('enables join game button when player name and game ID are entered', () => {
    render(<Home />);
    const button = screen.getByTestId('join-game-button');
    const playerInput = screen.getByTestId('player-name-input');
    const gameIdInput = screen.getByTestId('game-id-input');
    
    // Initially button should be disabled
    expect(button).toBeDisabled();
    
    // Enter player name only
    fireEvent.change(playerInput, { target: { value: 'Player1' } });
    expect(button).toBeDisabled();
    
    // Enter game ID as well
    fireEvent.change(gameIdInput, { target: { value: 'game123' } });
    expect(button).not.toBeDisabled();
  });
});
