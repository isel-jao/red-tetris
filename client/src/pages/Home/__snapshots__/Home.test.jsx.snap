// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page > renders correctly 1`] = `null`;
